File: app/lib/magazzino/models/montante_model.dart
// ignore_for_file: non_constant_identifier_names

import 'package:ows_test_v2/generale/models/abstract_model.dart';
import 'package:ows_test_v2/magazzino/models/corridoio_model.dart';

class MontanteModel implements AbstractModel {
  int id_montante;
  String? cod_montante;
  String? desc_montante;
  String? cod_alias;
  int? id_magazzino;
  int? id_zona;
  int? id_corridoio;
  DateTime? data_ins;
  DateTime? data_agg;

  CorridoioModel? corridoio;

  MontanteModel({
    required this.id_montante,
    this.cod_montante,
    this.desc_montante,
    this.cod_alias,
    this.id_magazzino,
    this.id_zona,
    this.id_corridoio,
    this.data_ins,
    this.data_agg,
    this.corridoio,
  });

  @override
  Map<String, dynamic>? toJson() {
    return <String, dynamic>{
      "id_montante": id_montante,
      "cod_montante": cod_montante,
      "desc_montante": desc_montante,
      "cod_alias": cod_alias,
      "id_magazzino": id_magazzino,
      "id_zona": id_zona,
      "id_corridoio": id_corridoio,
      "data_ins": data_ins,
      "data_agg": data_agg,
      "corridoio": corridoio?.toJson(),
    };
  }

  factory MontanteModel.fromJson(Map<String, dynamic> json) {
    return MontanteModel(
      id_montante: json["id_montante"],
      cod_montante: json["cod_montante"],
      desc_montante: json["desc_montante"],
      cod_alias: json["cod_alias"],
      id_magazzino: json["id_magazzino"],
      id_zona: json["id_zona"],
      id_corridoio: json["id_corridoio"],
      data_ins: DateTime.tryParse(json["data_ins"] ?? ''),
      data_agg: DateTime.tryParse(json["data_agg"] ?? ''),
      corridoio: json["corridoio"] != null ? CorridoioModel.fromJson(json["corridoio"]) : null,
    );
  }
}

File: web/Magazzino/Models/MontanteModel.cs
using System.Data.SqlClient;
using System.Security.Policy;
using web.Generale;

namespace web.Magazzino.Models
{
   
    public class MontanteModel : BasicModel
    {
        public int Id_montante { get; set; }
        public string? Cod_montante { get; set; }
        public string? Desc_montante { get; set; }
        public string? Cod_alias { get; set; }
        public int? Id_magazzino { get; set; }
        public int? Id_zona { get; set; }
        public int? Id_corridoio { get; set; }


        public CorridoioModel? Corridoio { get; set; }

        public MontanteModel()
        {
            NomeTabella = "montante";
            GestioneAzienda = false;
            GestioneLogUtenti = false;
        }

      

        public override void FromReader(SqlDataReader reader)
        {
            FromReaderCommon(reader);
            Id_montante = GetIntNonNullable(reader, nameof(Id_montante));
            Cod_montante = GetString(reader, nameof(Cod_montante));
            Desc_montante = GetString(reader, nameof(Desc_montante));
            Cod_alias = GetString(reader, nameof(Cod_alias));
            Id_magazzino = GetInt(reader, nameof(Id_magazzino));
            Id_zona = GetInt(reader, nameof(Id_zona));
            Id_corridoio = GetInt(reader, nameof(Id_corridoio));

            Corridoio = new CorridoioModel();
            Corridoio.FromReader(reader);
        }
    }
}
